cmd/template/framework/files/logger.go.tmpl

```diff
@@ -0,0 +1,94 @@
+package logger
+
+import (
+	"context"
+	"errors"
+	"log"
+	"log/slog"
+	"os"
+	"path/filepath"
+)
+
+type Logger struct {
+	innerLogger *slog.Logger
+	context     context.Context
+}
+
+type LoggerConfig struct {
+	LoggerType string
+	Level      slog.Level
+	AddSource  bool
+}
+
+// NewLogger creates a new Logger instance with the provided configuration.
+// It supports two types of loggers: "json" and "text" (passed as the LoggerType).
+// The function initializes a context (which is used internally only), sets up a handler based on the logger type,
+// and returns a pointer to the Logger struct.
+// The handler is responsible for formatting and outputting log messages.
+// If the configuration specifies to add source information to log messages,
+// the source file name will be extracted and only the base name will be used.
+// The function returns an error if an unsupported logger type is specified.
+func New(config LoggerConfig) (*Logger, error) {
+	ctx := context.Background()
+	var handler slog.Handler
+	replaceAttr := func(groups []string, a slog.Attr) slog.Attr {
+		if a.Key == "function" {
+			return slog.Attr{}
+		}
+
+		if config.AddSource && a.Key == slog.SourceKey {
+			source := a.Value.Any().(*slog.Source)
+			source.File = filepath.Base(source.File)
+		}
+
+		// if used in AWS or GCP (and Azure?), they'll add the timestamp themselves. Else uncomment to get timestamps
```

Comment:
It's better to enable this independently, even when used in AWS, GCP, or Azure, because the logger should be configured independently instead of depending on the `cloud provider` or whatever it is for timestamp handling.